specification.md

```diff
@@ -567,15 +602,24 @@ that the following state model is adhered to.
 Job instances are, in this API, stateful objects.  A job's state can be
 inspected via the `job.getStatus()` method which will return a `JobStatus`
 instance on which the job's state is available as an attribute.  State
-transitions can also be received via callbacks
+transitions can also be received via callbacks. The transitions through 
+`STAGE_IN`, `STAGE_OUT`, and `CLEANUP` must be generated even if a particular 
+job does not specify any files for staging or clean up.
 
 An implementation MUST ensure that job state transitions occur according to the
 following state model:  
   - A job is created in an initial state `NEW`.  
   - When the job is accepted by the backend for execution, it will enter the 
   state `QUEUED`.
+  - Before the job starts executing, it enters the 
```

Comment:
I would suggest 'before job _execution_ starts' (and 'ends' two lines below) to distinguish this from the job instance life time.

Reply:
Agreed. Thanks.

Reply:
Done